responses/test_responses.py

```diff
@@ -1527,76 +1568,106 @@ def run():
 
 
 def test_fail_request_error():
-    @responses.activate
+    """
+    Validate that exception is raised if request URL/Method/kwargs don't match
+    :return:
+    """
+
     def run():
-        responses.add("POST", "http://example1.com")
-        responses.add("GET", "http://example.com")
-        responses.add(
-            "POST",
-            "http://example.com",
-            match=[matchers.urlencoded_params_matcher({"foo": "bar"})],
-        )
-        responses.add(
-            "POST",
-            "http://example.com",
-            match=[matchers.json_params_matcher({"fail": "json"})],
-        )
+        with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
+            rsps.add("POST", "http://example1.com")
+            rsps.add("GET", "http://example.com")
 
-        with pytest.raises(ConnectionError) as excinfo:
-            requests.post("http://example.com", data={"id": "bad"})
+            with pytest.raises(ConnectionError) as excinfo:
+                requests.post("http://example.com", data={"id": "bad"})
 
-        msg = str(excinfo.value)
-        assert "- POST http://example1.com/ URL does not match" in msg
-        assert "- GET http://example.com/ Method does not match" in msg
+            msg = str(excinfo.value)
+            assert "- POST http://example1.com/ URL does not match" in msg
+            assert "- GET http://example.com/ Method does not match" in msg
 
-        if six.PY3:
-            assert "Parameters do not match. id=bad doesn't match {'foo': 'bar'}" in msg
-        else:
-            assert (
-                "Parameters do not match. id=bad doesn't match {u'foo': u'bar'}" in msg
-            )
+    run()
+    assert_reset()
 
-        assert (
-            "Parameters do not match. JSONDecodeError: Cannot parse request.body" in msg
-        )
 
-        # test query parameters
-        responses.add(
-            "GET",
-            "http://111.com",
-            match=[matchers.query_param_matcher({"my": "params"})],
-        )
+def test_fail_matchers_error():
+    """
+    Validate that Exception is raised if request does not match responses.matchers
+        validate matchers.urlencoded_params_matcher
+        validate matchers.json_params_matcher
+        validate matchers.query_param_matcher
+        validate matchers.request_kwargs_matcher
+    :return: None
+    """
 
-        responses.add(
-            method=responses.GET,
-            url="http://111.com/",
-            body="two",
-            match=[matchers.json_params_matcher({"page": "one"})],
-        )
+    def run():
+        with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
```

Comment:
Why did you move the tests to using `responses.RequestsMock()` over the module level methods?

Reply:
@markstory 
I wanted to limit the context.
Do you see any drawbacks of doing that?

Reply:
also, in this particular change I also separated tests that raise from matchers and from responses themselves.

Reply:
I don't think there are significant drawbacks to using `RequestsMock()`. The only one I can think of is that tests have multiple styles of using the library and it could be harder for future contributors and users to know which interface is the intended usage.

Reply:
if strongly required, I can do a change for response decorator

Reply:
Its not a strong requirement :smile: